splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/window/function/FirstLastValueFunction.java

```diff
@@ -60,7 +60,37 @@ protected void calculateOnAdd(WindowChunk chunk, DataValueDescriptor[] dvds) thr
 
     @Override
     protected void calculateOnRemove(WindowChunk chunk, DataValueDescriptor[] dvds) throws StandardException {
-       // nothing to do here we've already set all the results. For LAST_VALUE(), result is just the last item we added.
+       if (isLastValue) {
+           // we need to handle the scenario where the last value is the one that is removing, this could happen
+           // when ignoreNulls is true
+           DataValueDescriptor removedVal = dvds[0];
+           if (ignoreNulls && !removedVal.isNull() && chunk.getResult().equals(removedVal)) {
+               int position = chunk.last;
```

Comment:
Is the equals() check a reliable method to determine if this is the removed value?  Could there be a case where chunk.getResult() happens to equal the removed value, but there are multiple rows with the same value, and it's not really the one being removed from the window?

Reply:
Thanks @msirek for the comment. the equals() check does not need to ensure chunk.getResult() to be the same object as the removed value. It just needs to rule out the cases where the chunk.getResult() is not possible to be the same as the removed value to save some processing time. For example, 
       original chunk content:  3, null, null, null
       chunk.getResult() is 3 which is last value of the chunk ignore nulls.
after removing the first value in the chunk, we have:
       removed value: 3
       chunk content: null, null, null
The above logic will be triggered, and the updated chunk.getResult() will be null.

Another example:
       original chunk content:  3, null, 3, null
       chunk.getResult() is also 3, but not the first value in the chunk
after removing the first value in the chunk, we have
       removed value: 3
       chunk content: null, 3, null
The above logic will still be triggered, but the chunk.getResult() will still point to the 3 currently in the chunk after update.